src/mds/MDSRank.cc

```diff
@@ -330,12 +330,13 @@ class C_Drop_Cache : public MDSInternalContext {
         recall_client_state();
       });
       mds->timer.add_event_after(1.0, timer);
+      gather->activate();
     } else {
       if (!gather->has_subs()) {
-        delete gather;
+        gather->activate();
```

Comment:
wouldn't this assert in `C_GatherBuilderBase::activate()` since `finisher` is not set (yet)?

Reply:
No, see `::delete_me`. `Gather` does not require a finisher.

Reply:
umm -- I encountered the assert at this very point when implementing `C_Drop_Cache` which prompted me to use `delete`.

I'll recheck.

Reply:
Oops, you're right Venky: /ceph/teuthology-archive/pdonnell-2019-02-02_03:29:52-fs-wip-pdonnell-testing-20190202.003915-distro-basic-smithi/3538153/teuthology.log

Thanks!